test/new-e2e/tests/installer/upgrade_scenario_test.go

```diff
@@ -0,0 +1,213 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package installer
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/tests/installer/host"
+	e2eos "github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/stretchr/testify/require"
+)
+
+type upgradeScenarioSuite struct {
+	packageBaseSuite
+}
+
+type packageEntry struct {
+	Package string `json:"package"`
+	Version string `json:"version"`
+	URL     string `json:"url"`
+}
+
+type catalog struct {
+	Packages []packageEntry `json:"packages"`
+}
+
+type packageStatus struct {
+	State             string `json:"state"`
+	StableVersion     string `json:"stable_version"`
+	ExperimentVersion string `json:"experiment_version"`
+}
+
+var testCatalog = catalog{
+	Packages: []packageEntry{
+		{
+			Package: "datadog-agent",
+			Version: latestAgentImageVersion,
+			URL:     fmt.Sprintf("oci://gcr.io/datadoghq/agent-package:%s", latestAgentImageVersion),
+		},
+	},
+}
+
+// datadog-agent
+//
+//	State: OK
+//	Installed versions:
+//	  ● stable: v7.52.0-rc.1.git.15.6c19b17.pipeline.28815219-1
+//	  ● experiment: none
+var installerStatusRegex = regexp.MustCompile(`([a-zA-Z-]+)\n[ \t]+State:.([a-zA-Z-]+)\n[ \t]+Installed versions:\n[ \t]+..stable:.([a-zA-Z0-9-\.]+)\n[ \t]+..experiment:.([a-zA-Z0-9-\.]+)`)
```

Comment:
TODO next PR: move this in the host subdir and use the raw API to get JSON instead of a formatted message that will likely break